api/src/main/java/org/pmiops/workbench/cdr/dao/CriteriaDao.java

```diff
@@ -1,11 +1,14 @@
 package org.pmiops.workbench.cdr.dao;
 
 import org.pmiops.workbench.cdr.model.Criteria;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 
 import java.util.List;
 
 public interface CriteriaDao extends CrudRepository<Criteria, Long> {
 
-    List<Criteria> findCriteriaByTypeLikeAndParentIdOrderBySortOrderAsc(String type, Long parentId);
+    @Query("select c from Criteria c where c.type like :type% and c.parentId = :parentId order by c.sortOrder asc")
```

Comment:
So did TypeLike not work?

Remind me why we need to use "LIKE" at all? This is like prefix matching on concepts?

Reply:
Demographics tree has different types.. DEMO_AGE, DEMO_RACE, DEMO_DEC and DEMO_GEN. To get the tree i select where like 'DEMO%'

Reply:
type like didn't add the % to the param, so i addded the named parameter :type%

Reply:
OK. You may be able to do findCriteriaByTypeStartingWithAndParentIdOrderBySortOrderAsc(). See:

https://stackoverflow.com/questions/21456494/spring-jpa-query-with-like

Does the type column get used for anything other than matching the "DEMO" prefix? (You could just have a criteria type enum field with values like DEMO, ICD9, etc. and query on that... and then user parent_id for groups within it...)

Reply:
the UI makes request for counts using the type column from the criteria, so DEMO has the 4 types to differentiate from... mainly cause the sql is different for each of the 4 types of DEMO.